src/workerd/server/server.c++

```diff
@@ -1228,6 +1228,8 @@ public:
           };
 
           TimerChannel& timerChannel = service;
+
+          Worker::Lock lock(*service.worker, asyncLock);
```

Comment:
I guess moving the lock to this scope increases the amount of work done while holding the lock a little, but I assume it's nothing to worry about, particularly if this is just setting up for additional initialization to come.

Reply:
Yeah, I think it should be okay. It's pretty much just holding it for an extra allocation, which should be pretty cheap with tcmalloc.